### ISSUE TITLE
Suggested code pattern for Android triggers Android Studio warning

### ISSUE THREAD

**Issue by ewoerner** [kalinjul/kotlin-multiplatform-oidc#71]
The suggested pattern for the AndroidCodeAuthFlowFactory singleton triggers a warning in Android studio:
![image](https://github.com/user-attachments/assets/a260efdc-c416-4f98-a84a-62ad88428534)

**Comment by kalinjul** [comment]
Yes, please keep an instance of the factory using your favourite Dependency Injection framework as suggested in the Readme. I will however remove the companion object to not irritate users.
